features/device-config/service/src/main/java/org/opennms/features/deviceconfig/service/impl/DeviceConfigServiceImpl.java

```diff
@@ -112,20 +114,69 @@ public void triggerConfigBackup(String ipAddress, String location, String config
                 });
     }
 
-    private CompletableFuture<PollerResponse> pollDeviceConfig(String ipAddress, String location, String configType) {
-        if (Strings.isNullOrEmpty(configType)) {
-            configType = ConfigType.Default;
-        }
-        String serviceName = DEVICE_CONFIG_SERVICE_NAME_PREFIX + configType;
-        MonitoredService service = sessionUtils.withReadOnlyTransaction(() -> {
+    @Override
+    public List<RetrivalConfiguration> getRetrivalConfigurations(final String ipAddress, final String location) {
+        final var iface = this.ipInterfaceDao.findByIpAddressAndLocation(ipAddress, location);
+
+        return iface
+                // Get all device config services defined for this interface
+                .getMonitoredServices().stream()
+
+                // Resolve the service name into service config
+                .flatMap(svc -> this.pollerConfig.findService(InetAddressUtils.str(svc.getIpAddress()), svc.getServiceName()).stream())
+
+                // Filter for the device config monitor
+                .filter(match -> this.pollerConfig.getServiceMonitor(match.service.getName()).getClass() == DeviceConfigMonitor.class)
```

Comment:
This brings direct dependency on DeviceConfigMonitor for the service.  
One messy way is to get Monitors by className from `PollerRequestBuilderImpl` just like you did with `getInterpolatedAttributes()` and match the monitor.

Reply:
Any reason why we should avoid this dependency? Having it makes it save for refactoring.

I can't see how the `PollerRequestBuilder` could help here. This is about resolving the service name to the assigned monitoring class, which is only available from poller config.